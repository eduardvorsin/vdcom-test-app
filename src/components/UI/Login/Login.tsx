import React from 'react';
import { useActionData } from 'react-router-dom';
import { Fade, AlertTitle, Alert, Box } from '@mui/material';
import LoginForm from '../LoginForm/LoginForm';

const Login = () => {
  const actionError = useActionData();
  const error = actionError instanceof Error ? actionError : '';

  return (
    <Box data-testid='login'>
      {error && (
        <Fade
          in={Boolean(error)}
        >
          <Alert
            severity='error'
            variant='standard'
            sx={{
              mb: '30px',
            }}
          >
            <AlertTitle>Error</AlertTitle>
            {error.message}
          </Alert>
        </Fade>
      )}

      <LoginForm />
    </Box>
  );
};

export default Login;
