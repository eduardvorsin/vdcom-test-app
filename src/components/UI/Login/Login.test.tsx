import React from 'react';

import { render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import Login from './Login';

const routes = [{
  path: '/login',
  element: <Login />,
}];

const router = createMemoryRouter(routes, {
  initialEntries: ['/login'],
});

describe('Login tests', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('renders correctly', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByTestId<HTMLDivElement>('login')).toBeInTheDocument();
    expect(screen.getByLabelText<HTMLInputElement>(/Login/i)).toBeInTheDocument();
    expect(screen.getByLabelText<HTMLInputElement>(/Password/i)).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByTestId<HTMLDivElement>('login')).toMatchSnapshot();
  });
});
