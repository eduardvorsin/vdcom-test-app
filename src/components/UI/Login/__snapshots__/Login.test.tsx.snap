// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests is a basic snapshot 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="login"
>
  <form
    action="/login"
    data-testid="login-form"
    method="post"
    novalidate=""
  >
    <div
      class="MuiBox-root css-utmfdt"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17c6y5f-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            Login
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="username"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              name="username"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Login
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17c6y5f-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r3:"
            id=":r3:-label"
          >
            Password
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r3:"
              name="password"
              required=""
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Password
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-tddfjc"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-kcn0n4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Sign in
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
